packages/devtools/src/ts/devtoolsPanel/messaging.ts

```diff
@@ -27,80 +43,122 @@ const actionHistory$ = new Observable<ActionHistory>(obs => {
     backgroundPageConnection.disconnect()
     copySubscription.unsubscribe()
   }
-}).pipe(
-  startWith([]),
-  shareLatest(),
-);
+}).pipe(startWith([]), shareLatest())
 
-type Action = ActionHistory extends Array<infer R> ? R : never;
+type Action = ActionHistory extends Array<infer R> ? R : never
 
-const tagStateReducer = (state: Record<string, DebugTag> = {}, action: Action): Record<string, DebugTag> => {
-  switch(action.type) {
-    case 'newTag$':
-      return {
-        ...state,
-        [action.payload.id]: {
-          ...action.payload,
-          refs: [],
-          latestValues: {},
-        },
-      };
-    case 'tagRefDetection$':
+const tagStateReducer = (
+  state: DebugTag | null,
+  action: Action,
+): DebugTag | null => {
```

Comment:
I changed this one to track a single tag. It should've been broken down like this previously 🙈